### ISSUE TITLE
Maybe corrupted .m4a when downloading livestreams

### ISSUE THREAD

**Issue by computerfreak105** [gpodder/gpodder#1646]
Please help me solve a problem which I have for a long time.
When I download livestreams on youtube with yt-dlp directly with the format option (140/m4a) everything is ok and I can play the files fully without problem.
However when GPodder downloads youtube livestreams I get .m4a files which may take up to 20 seconds until they start playing with MPC-HC Media Player, or they play only a few seconds or not at all, par example with Foobar 2000. VLC Media also takes longer until such files begin to play. Very long streams above 4 hours aren't played fully, but when I convert them to .mp3 everything is ok again.

Here is a gamer video of 1 hour:

https://www.youtube.com/watch?v=4KV3O0TIVuU

And here is a link containing this livestream in audio. I downloaded it with GPodder and with yt-dlp. The GPodder-File is larger and does play only about 1 minute in Foobar2000, takes longer to open in VLC Media player. The yt-dlp version hasn't got this problem.

https://wetransfer.com/downloads/fa609332a7486d71c849f069a158a09e20240801154752/34dc1a6d6ecbaa5894a142fcb356403520240801154830/365c4f

The full command for downloading youtube channels with yt-dlp is:

d:\yt-dlp.exe --min-filesize 1m --geo-bypass --no-part -f 140/m4a --ignore-errors -o D:\yt/%(uploader)s/%(title)s_%(upload_date)s._%(autonumber)s.%(ext)s -a D:\YtUrl.txt

And my download options in gpodder are:

 "youtube": {
    "preferred_fmt_id": 0,
    "preferred_fmt_ids": [
      251,
      250,
      249,
      140
    ],
    "preferred_hls_fmt_id": 0,
    "preferred_hls_fmt_ids": [
      91,
      92,
      93,
      94,
      95,
      96
    ]
  }
}

Yt-dlp extension manages downloads but not channels.

I'd like to know why livestreams are downloaded differently in GPodder and if there is a solution for this.

Thanks for your help.

**Comment by computerfreak105** [comment]
A more permanent link to my files

https://www.dropbox.com/scl/fo/drw9qocakmqf6ezmtjb32/AAxxV_rRjsLJiai6WZHpOFo?rlkey=lcv3bk82tv7g9bbj0w6aeygeb&dl=0

**Comment by auouymous** [comment]
Your gpodder file is `ISO Media, MPEG v4 system, Dynamic Adaptive Streaming over HTTP`, otherwise known as DASH. I'm not sure why some videos would work while others don't, but you might need ffmpeg to turn the DASH into an M4A. I have ffmpeg and all of the downloads are working m4a files with format 140 in gpodder.

**Comment by auouymous** [comment]
Install homebrew, then install ffmpeg with it. I don't have/use a Mac and can't help more than that.

The problem is that gpodder's launcher on Mac does something to the PATH so it can't see ffmpeg (#1638), and requires you to modifiy the source code. You must manually modify gpodder to use it, see https://github.com/gpodder/gpodder/issues/1634#issuecomment-2184125074. I am going to try getting the Mac build working today, and if that happens, I will add ffmpeg_location to Settings.json.

```
        self._ydl_opts = {
            'cachedir': cachedir,
            'noprogress': True,  # prevent progress bar from appearing in console
            'ffmpeg_location': 'Where/I/Had/My/FFMPEG/File',
        }
```

The value is probably `/opt/homebrew/bin/ffmpeg` if you install it with homebrew.

**Comment by auouymous** [comment]
If you get ffmpeg installed, the Mac build at https://github.com/gpodder/gpodder/actions/runs/10275061323/artifacts/1783269494 adds a `path.alternate` field to Settings.json. After installing, run and quit gpo to add the field, then find that setting and add the directory ffmpeg is in, such as '/opt/homebrew/bin', in the empty single quotes.

See https://github.com/gpodder/gpodder/pull/1647#issuecomment-2272727722 for more details.

**Comment by computerfreak105** [comment]
Thank you for your instructions, I just installed it and will do a test overnight.

 

 

 

Von: auouymous ***@***.***> 
Gesendet: Mittwoch, 7. August 2024 09:01
An: gpodder/gpodder ***@***.***>
Cc: computerfreak105 ***@***.***>; Author ***@***.***>
Betreff: Re: [gpodder/gpodder] Maybe corrupted .m4a when downloading livestreams (Issue #1646)

 

If you get ffmpeg installed, the Mac build at https://github.com/gpodder/gpodder/actions/runs/10275061323/artifacts/1783269494 adds a path.alternate field to Settings.json. After installing, run and quit gpo to add the field, then find that setting and add the directory ffmpeg is in, such as '/opt/homebrew/bin', in the empty single quotes.

See #1647 (comment) <https://github.com/gpodder/gpodder/pull/1647#issuecomment-2272727722>  for more details.

**Comment by computerfreak105** [comment]
@Ploddles
I looked up the path of ffmpeg installed with homebrew:

/opt/homebrew/Cellar/ffmpeg/7.0.2/bin/ffmpeg

So I adjusted the settings.json to 

    "alternate": "'/opt/homebrew/Cellar/ffmpeg/7.0.2/bin/ffmpeg'"

Is this correct? I didn't get the message to allow GPodder to access ffmpeg, so I'm not sure I set the path correctly.


And @auouymous
When I start GPO I get some error messages, although the program is working so far when I use gpodder GUI.
However I wanted to provide more information and tried to debug from the terminal with /Applications/gPodder.app/Contents/MacOS/gpodder -v

This got me the message that GPodder was closed unexpected and I could generate a problem report for apple. I didn't send one, but I created it. I send you two files to inspect. Hope this helps in solving potential issues.
[logs.zip](https://github.com/user-attachments/files/16538575/logs.zip)

**Comment by auouymous** [comment]
@computerfreak105 You wrapped the directory name in two sets of quotes, get rid of the single or the double.

**Comment by Ploddles** [comment]
I don’t use the home-brew version. I use the one by the yt-dlp team as they update it to keep it fully compatible with yt-dlp.

https://github.com/yt-dlp/FFmpeg-Builds￼
yt-dlp/FFmpeg-Builds: FFmpeg Builds for yt-dlp
github.com

**Comment by auouymous** [comment]
@Ploddles Thank you for that ffmpeg link, I'll add it to the website. And thank you for testing gpo.

@computerfreak105 I will need to figure out how to fix gpo, I'm not sure why the gtk client can find glib but gpo can't.

**Comment by computerfreak105** [comment]
@Ploddles <https://github.com/Ploddles> 

 

There doesn’t seem to be a mac version of FFmpeg Builds for yt-dlp, right? Do you just take a linux version then? I have a Mac with M1, which one should I take? 

 

And would I just copy the ffmpeg, ffplay and ffprobe from the bin folder into programs and set the path of settings.json to ffmpeg in applications, or are there other steps to take like with yt-dlp and copying files into the gpodder folder?



 

 

 

 

 

 

Von: Ploddles ***@***.***> 
Gesendet: Donnerstag, 8. August 2024 10:28
An: gpodder/gpodder ***@***.***>
Cc: computerfreak105 ***@***.***>; Mention ***@***.***>
Betreff: Re: [gpodder/gpodder] Maybe corrupted .m4a when downloading livestreams (Issue #1646)

 

I don’t use the home-brew version. I use the one by the yt-dlp team as they update it to keep it fully compatible with yt-dlp. 

https://github.com/yt-dlp/FFmpeg-Builds￼ 
yt-dlp/FFmpeg-Builds: FFmpeg Builds for yt-dlp 
github.com

**Comment by Ploddles** [comment]
> @Ploddles <https://github.com/Ploddles> There doesn’t seem to be a mac version of FFmpeg Builds for yt-dlp, right? Do you just take a linux version then? I have a Mac with M1, which one should I take? And would I just copy the ffmpeg, ffplay and ffprobe from the bin folder into programs and set the path of settings.json to ffmpeg in applications, or are there other steps to take like with yt-dlp and copying files into the gpodder folder? 

Just use the latest Linux one. Unless you specifically need ffplay & ffprobe just copy ffmpeg there. 

Once you have done that, right click it and open it. macOS will then prompt you to Allow it to run or not, so Allow it. Check Security in System Preferences and Allow it there as well. It should then open a Terminal windows which you can just close and you are all set.

**Comment by computerfreak105** [comment]
<https://github.com/Ploddles> @Ploddles

 

Thanks, I installed the linux version for ARM, since I have an M1 Mac. I coppied ffmpeg to the Applications and opened it with right click. I allowed it to open.

 

But I didn’t find anything in system preferences and security to allow. And ffmpeg apparently didn’t want file and folder access either.

 

 

May I ask what your alternate path looks like in the settings.json file, please? I tried with

    "alternate": "/Applications/ffmpeg"

 

But the .m4a livestreams still didn’t play as other non live .m4a files.

 

Then I tried with 

'Applications/ffmpeg'

But my whole settings.json got reverted back to the original because of that.

 

So, I’m currious how you’re doing it and what your path looks like.

 

 

Thanks!

 

Von: Ploddles ***@***.***> 
Gesendet: Freitag, 9. August 2024 12:54
An: gpodder/gpodder ***@***.***>
Cc: computerfreak105 ***@***.***>; Mention ***@***.***>
Betreff: Re: [gpodder/gpodder] Maybe corrupted .m4a when downloading livestreams (Issue #1646)

 

@Ploddles <https://github.com/Ploddles>  https://github.com/Ploddles There doesn’t seem to be a mac version of FFmpeg Builds for yt-dlp, right? Do you just take a linux version then? I have a Mac with M1, which one should I take? And would I just copy the ffmpeg, ffplay and ffprobe from the bin folder into programs and set the path of settings.json to ffmpeg in applications, or are there other steps to take like with yt-dlp and copying files into the gpodder folder?

Just use the latest Linux one. Unless you specifically need ffplay & ffprobe just copy ffmpeg there.

Once you have done that, right click it and open it. macOS will then prompt you to Allow it to run or not, so Allow it. Check Security in System Preferences and Allow it there as well. It should then open a Terminal windows which you can just close and you are all set.

**Comment by Ploddles** [comment]
@computerfreak105 

You only want the path to the directory it is in, so not "/Applications/ffmpeg" but just "/Applications", unless you have created a folder called ffmpeg and copied the file ffmpeg into it.

I just use the GUI rather than manually changing the json file and it's just in my Downloads folder at the moment.

You may need to add Macintosh HD to the path, so "Macintosh HD/Applications". 

I'd also try the x86 version rather than the Arm version.

<img width="748" alt="Screenshot 2024-08-09 at 22 54 24" src="https://github.com/user-attachments/assets/99643c53-d5d5-4e40-adbd-afa8c85236e9">

**Comment by computerfreak105** [comment]
@Ploddles <https://github.com/Ploddles> 

 

Thanks, now I run it from the download folder like you, but I have no luck yet. I took the 64-bit linux version since I couldn’t find a 86x one. I gave ffmpeg and gpodder full access to my harddrive under system preferences and security. But I wonder if these two are really working together at all.

 

Could you maybe provide me a link to the ffmpeg which is working for you?

 

And how did you add ffmpeg in the GUI? It’s not very accessible for visually impaired users because foundamental keyboard navigation isn’t possible in many cases. But I’d like to try anyway. And if I can’t do it maybe some sighted people of my workplace can do it for me.

 

 

 

 

 

Von: Ploddles ***@***.***> 
Gesendet: Samstag, 10. August 2024 00:01
An: gpodder/gpodder ***@***.***>
Cc: computerfreak105 ***@***.***>; Mention ***@***.***>
Betreff: Re: [gpodder/gpodder] Maybe corrupted .m4a when downloading livestreams (Issue #1646)

 

@computerfreak105 <https://github.com/computerfreak105> 

You only want the path to the directory it is in, so not "/Applications/ffmpeg" but just "/Applications", unless you have created a folder called ffmpeg and copied the file ffmpeg into it.

I just use the GUI rather than manually changing the json file and it's just in my Downloads folder at the moment.

You may need to add Macintosh HD to the path, so "Macintosh HD/Applications".

I'd also try the x86 version rather than the Arm version.

Screenshot.2024-08-09.at.22.54.24.png (view on web) <https://github.com/user-attachments/assets/99643c53-d5d5-4e40-adbd-afa8c85236e9>

**Comment by auouymous** [comment]
Fixed by #1649.
